packages/vm/vmtxbuilder/txbuilder.go

```diff
@@ -145,3 +145,7 @@ func (txb *AnchorTransactionBuilder) BuildTransactionEssence(stateMetadata []byt
 	}
 	return ptb.Finish()
 }
+
+func (txb *AnchorTransactionBuilder) ViewPTB() *iotago.ProgrammableTransactionBuilder {
+	return txb.ptb.Clone()
```

Comment:
There is a lot of stuff that is added to the PTB when `BuildTransactionEssence` is called. That is not being included here? In that case the estimation will be always lower than the real size.

Look at the stardast implementation:

https://github.com/iotaledger/wasp/blob/06b7a38d2e1360fc4f9a5fadf4288383121e0b09/packages/vm/vmimpl/runreq.go#L514-L521

Reply:
I reformed the way to check the tx size. I build the tx every time the check func is called